### ISSUE TITLE
Cloudflare Stream is unsupported

### ISSUE THREAD

**Issue by emv33** [yt-dlp/yt-dlp#9171]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Spain

### Example URLs

https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe

### Provide a description that is worded well enough to be understood

Says URL is unsupported.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU 'https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe'
[debug] Command-line config: ['-vU', 'https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.02.05.232712 from yt-dlp/yt-dlp-nightly-builds [05420227a] (pip)
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2023.11.17, mutagen-1.47.0, requests-2.31.0, secretstorage-3.3.1, sqlite3-3.37.2, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1832 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.02.05.232712 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.02.05.232712 from yt-dlp/yt-dlp-nightly-builds)
[generic] Extracting URL: https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe
[generic] iframe: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] iframe: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1586, in wrapper
    return func(self, *args, **kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1721, in __extract_info
    ie_result = ie.extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 718, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/generic.py", line 2531, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe
```

**Comment by bashonly** [comment]
this would fix it
```diff
diff --git a/yt_dlp/extractor/cloudflarestream.py b/yt_dlp/extractor/cloudflarestream.py
index 748e8e908..01e57662b 100644
--- a/yt_dlp/extractor/cloudflarestream.py
+++ b/yt_dlp/extractor/cloudflarestream.py
@@ -10,7 +10,7 @@ class CloudflareStreamIE(InfoExtractor):
     _VALID_URL = r'''(?x)
                     https?://
                         (?:
-                            (?:watch\.)?%s/|
+                            (?:[\w-]+\.)?%s/|
                             %s
                         )
                         (?P<id>%s)
```

**Comment by bashonly** [comment]
related, and could possibly be solved by the same PR: #7858

**Comment by seproDev** [comment]
@ilikenwf Your workaround was the first comment in this issue. yt-dlp will detect `customer-...` URLS once https://github.com/yt-dlp/yt-dlp/pull/9280 is merged.

**Comment by bashonly** [comment]
> So say you want to watch the TCN Putin interview

there is no cloudflarestream iframe in the html

**Comment by ilikenwf** [comment]
> > So say you want to watch the TCN Putin interview
> 
> there is no cloudflarestream iframe in the html

Just as with your upride.cc site, there is...but if it's loading via JS we are out of luck... but right clicking once loaded in browser and opening the iFrame works too...however then you have to still change the url from customer-whatever to watch, I think this is where the extractor's regex may need adjusted?

**Comment by bashonly** [comment]
the tucker clownshow is loading via JS

**Comment by ilikenwf** [comment]
All these cloudflare iframe embeds tend to be "customer-90230t2930.cloudflarestream.com" and our extractor doesn't play nice with those, only with the watch subdomains...so either we need to recognize the customer-whatever subdomains in our regex or rewrite them to work correctly.

Just because you disagree with Tucker and/or Putin's politics doesn't make the fact that the dictator of Russia did an interview any less important, making it historically significant. Can't throw out the baby with the bathwater kind of thing...consuming all your news from one outlet or source means that you accept the bias of whatever news outlet as truth...gotta consume info from all sides to make a fair evaluation of reality.

**Comment by bashonly** [comment]
Ok sure but this is off-topic anyways, #9169

**Comment by ilikenwf** [comment]
No sweat, I don't take it personally, I just try to be pragmatic with how I build my worldview.

Yes, but in reference to cloudflare stream - even with the UpRide.cc, Tucker Carlson, and literally any other cloudflarestream embed...take any of them and change the iframe URI to watch.cloudflarestream.com and it'll work, so we need to modify the regex.

**Comment by ilikenwf** [comment]
> @ilikenwf Your workaround was the first comment in this issue. yt-dlp will detect `customer-...` URLs once #9280 is merged.

Except that doesn't work in my testing. I tested with 

https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe%5C?autoplay%5C=true%5C&letterboxColor%5C=transparent

And had no success.

**Comment by seproDev** [comment]
> once https://github.com/yt-dlp/yt-dlp/pull/9280 is merged.

When I wrote that, the PR wasn't merged. Now it is merged. If you use the master build generated in a couple of minutes it will work.

**Comment by ilikenwf** [comment]
I tested with it, it doesn't work.

**Comment by bashonly** [comment]
```
[debug] Command-line config: ['-vF', 'https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe%5C?autoplay%5C=true%5C&letterboxColor%5C=transparent']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34]
[debug] Lazy loading extractors is disabled
[debug] Python 3.11.7 (CPython x86_64 64bit) - Linux-6.7.6-arch1-1-x86_64-with-glibc2.39 (OpenSSL 3.2.1 30 Jan 2024, glibc 2.39)
[debug] exe versions: ffmpeg 6.1.1 (setts), ffprobe 6.1.1
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.02.02, mutagen-1.47.0, requests-2.31.0, secretstorage-3.3.3, sqlite3-3.45.1, urllib3-2.2.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1833 extractors
[CloudflareStream] Extracting URL: https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/2463f6d3e06fa29710a337f5f5389fd8/iframe%5C?autoplay%5C=true%5C&letterboxColor%5C=transparent
[CloudflareStream] 2463f6d3e06fa29710a337f5f5389fd8: Downloading m3u8 information
[CloudflareStream] 2463f6d3e06fa29710a337f5f5389fd8: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 2463f6d3e06fa29710a337f5f5389fd8:
ID                       EXT RESOLUTION FPS │   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
hls-group_audio-original mp4 audio only     │       m3u8  │ audio only        unknown             [en-b60319a5] original
dash-638557390           m4a audio only     │  135k dash  │ audio only        mp4a.40.2  135k 48k [original] DASH audio, m4a_dash
dash-638556837           mp4 426x240        │  843k dash  │ avc1.42c015  843k video only          DASH video, mp4_dash
hls-283                  mp4 426x240     30 │  283k m3u8  │ avc1.42c015  283k video only
dash-638557133           mp4 640x360        │ 1076k dash  │ avc1.4d401e 1076k video only          DASH video, mp4_dash
hls-353                  mp4 640x360     30 │  354k m3u8  │ avc1.4d401e  354k video only
dash-638557644           mp4 852x480        │ 1505k dash  │ avc1.4d401f 1505k video only          DASH video, mp4_dash
hls-468                  mp4 852x480     30 │  468k m3u8  │ avc1.4d401f  468k video only
dash-638561817           mp4 1280x720       │ 2543k dash  │ avc1.4d401f 2543k video only          DASH video, mp4_dash
hls-824                  mp4 1280x720    30 │  824k m3u8  │ avc1.4d401f  824k video only
dash-638568841           mp4 1920x1080      │ 4801k dash  │ avc1.4d4028 4801k video only          DASH video, mp4_dash
hls-2224                 mp4 1920x1080   30 │ 2225k m3u8  │ avc1.4d4028 2225k video only
```

**Comment by ilikenwf** [comment]
Interesting. I'll update again and try again.

**Comment by paul1149** [comment]
I'm on YT-DLP 24.5.23 and it's not working here either.

```https://customer-aw5py76sw8wyqzmh.cloudflarestream.com/embed/sdk-iframe-integration.fla9.latest.js?video=0e8e040aec776862e1d632a699edf59e```

```$ yt-dlp https://watch.cloudflarestream.com/embed/sdk-iframe-integration.fla9.latest.js?video=0e8e040aec776862e1d632a699edf59e
[generic] Extracting URL: https://watch.cloudflarestream.com/embed/sdk-iframe-integration.fla9.latest.js?video=0e8e040aec776862e1d632a699edf59e
[generic] sdk-iframe-integration.fla9.latest: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] sdk-iframe-integration.fla9.latest: Extracting information
[CloudflareStream] Extracting URL: https://embed.cloudflarestream.com/embed/r4xu.fla9.latest.js?video=sdk-iframe-integration.fla9.latest
ERROR: [CloudflareStream] sdk-iframe-integration.fla9.latest: sdk-iframe-integration.fla9.latest: Failed to parse JSON (caused by UnicodeDecodeError('utf-8', b'~V\xbd', 2, 3, 'invalid start byte'));  ```

**Comment by bashonly** [comment]
@paul1149 open a new issue with a complete **verbose** log
